pkg/chunk/cassandra/storage_client.go

```diff
@@ -97,11 +102,23 @@ func (cfg *Config) setClusterConfig(cluster *gocql.ClusterConfig) {
 		}
 	}
 	if cfg.Auth {
+		if cfg.Password != "" && cfg.PasswordFile != "" {
+			return errors.Errorf("The -cassandra.password and -cassandra.password-file flags are mutually exclusive.")
+		}
+		password := cfg.Password
+		if password == "" {
+			passwordBytes, err := ioutil.ReadFile(cfg.PasswordFile)
+			if err != nil {
+				return errors.Errorf("Could not read Cassandra password file: %v", err)
+			}
+			password = string(passwordBytes)
```

Comment:
I would suggest to trim the whitespaces from the file, otherwise a newline at the end of the file will break it and may be annoying to detect. You can trim whitespaces with `strings.TrimSpace()`

Reply:
I think remove `\n` is reasonable, but what if the password contains space?

Reply:
Your solution using `TrimRight()` looks good to me.